lib/phoenix_ecto/html.ex

```diff
@@ -231,17 +231,21 @@ if Code.ensure_loaded?(Phoenix.HTML) do
     defp form_for_errors(%{action: nil}), do: []
     defp form_for_errors(%{errors: errors}), do: errors
 
-    defp form_for_hidden(data) do
+    defp form_for_hidden(%{__struct__: _} = data) do
       # Since they are primary keys, we should ignore nil values.
       for {k, v} <- Ecto.primary_key(data), v != nil, do: {k, v}
     end
+    defp form_for_hidden(_), do: []
 
     defp form_for_name(%{__struct__: module}) do
       module
       |> Module.split()
       |> List.last()
       |> Macro.underscore()
     end
+    defp form_for_name(_) do
+      raise ArgumentError, "non-struct data passed in the changeset without `:as` option set"
```

Comment:
This tell what went wrong but it doesn't tell what the user should do. Maybe we should rephrase it to something like "non-struct data in changeset requires the `:as` option to be given"?